mis_builder/models/mis_builder.py

```diff
@@ -0,0 +1,865 @@
+                                               ids, ['field_ids'], 20), }),
+        'date_field': fields.many2one('ir.model.fields', required=True,
+                                      string='Date field',
+                                      domain=[('ttype', 'in',
+                                               ('date', 'datetime'))]),
+        'domain': fields.char(string='Domain'),
+        'report_id': fields.many2one('mis.report', string='Report',
+                                     ondelete='cascade'),
+    }
+
+    _order = 'name'
+
+    def _check_name(self, cr, uid, ids, context=None):
+        for record_name in self.read(cr, uid, ids, ['name']):
+            if not _is_valid_python_var(record_name['name']):
+                return False
+        return True
+
+    _constraints = [
+        (_check_name, 'The name must be a valid python identifier', ['name']),
+    ]
+
+
+class mis_report(orm.Model):
+
+    """ A MIS report template (without period information)
+
+    The MIS report holds:
+    * an implicit query fetching all the account balances;
+      for each account, the balance is stored in a variable named
+      bal_{code} where {code} is the account code
+    * an implicit query fetching all the account balances solde;
+      for each account, the balance solde is stored in a variable named
+      bals_{code} where {code} is the account code
+    * a list of explicit queries; the result of each query is
+      stored in a variable with same name as a query, containing as list
+      of data structures populated with attributes for each fields to fetch
+    * a list of KPI to be evaluated based on the variables resulting
+      from the balance and queries
+    """
+
+    _name = 'mis.report'
+
+    _columns = {
+        'name': fields.char(size=32, required=True,
+                            string='Name', translate=True),
+        'description': fields.char(required=False,
+                                   string='Description', translate=True),
+        'query_ids': fields.one2many('mis.report.query', 'report_id',
+                                     string='Queries'),
+        'kpi_ids': fields.one2many('mis.report.kpi', 'report_id',
+                                   string='KPI\'s'),
+    }
+
+    # TODO: kpi name cannot be start with query name
+
+    def create(self, cr, uid, vals, context=None):
+        # TODO: explain this
+        if 'kpi_ids' in vals:
+            mis_report_kpi_obj = self.pool.get('mis.report.kpi')
+            for idx, line in enumerate(vals['kpi_ids']):
+                if line[0] == 0:
+                    line[2]['sequence'] = idx + 1
+                else:
+                    mis_report_kpi_obj.write(
+                        cr, uid, [line[1]], {'sequence': idx + 1},
+                        context=context)
+        return super(mis_report, self).create(cr, uid, vals, context=context)
+
+    def write(self, cr, uid, ids, vals, context=None):
+        # TODO: explain this
```

Comment:
https://github.com/odoo/odoo/blob/8.0/addons/account/account_bank_statement.py#L44
it is use to correctly update sequence